File: ios_client/Restauranties/Models/Restaurant.swift
//
//  Restaurant.swift
//  Restauranties
//
//  Created by Karim Alweheshy on 11/2/20.
//

import Foundation

struct Restaurant: Codable, Hashable {
    let id: String
    let name: String
    let phone: String?
    let imageURL: URL?
    let categories: [String]?
    let ownerID: String
    let totalRatings: Int
    let averageRating: Double
    let creationDate: Date
    let modificationDate: Date
}

File: firebase_cloud_functions/functions/src/models/restaurant.ts
export default class Restaurant {
    name: string
    phone: string
    imageURL: URL
    categories: string[]
    ownerID: string
    totalRatings: number
    averageRating: number
    noReplyCount: number

    constructor(
        name: string,
        phone: string,
        imageURL: URL,
        categories: string[],
        ownerID: string,
        totalRatings: number,
        averageRating: number,
        noReplyCount: number
    ) {
        this.name = name
        this.phone = phone
        this.imageURL = imageURL
        this.categories = categories
        this.ownerID = ownerID
        this.totalRatings = totalRatings
        this.averageRating = averageRating
        this.noReplyCount = noReplyCount
    }
}